samples: rtc: ds1307: Implementation of RTC API

walidbadar opened the pull request:
This sample application periodically reads date and time data from the DS1307 RTC module, and displays on the console.

Tested with esp32_devkitc_wroom board and DS1307 RTC module.

walidbadar committed:
drivers: rtc: ds1307: Implementation of RTC API

This sample application periodically reads date and time data from the DS1307 RTC module, and displays on the console.

Signed-off-by: Muhammd Waleed Badar <walid.badar@gmail.com>

faxe1008 reviewed (changes requested):
~~There is already a driver for this intree:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers%2Frtc%2Frtc_ds1307.c~~

@walidbadar  Nevermind I misinterpreted the title of the pull request. A general RTC sample would be neat, if there is a board that has an RTC already included. Or maybe just a `build_all` test for them, as some of the drivers are not tested by CI.

faxe1008 commented on samples/drivers/rtc/ds1307/LICENSE:
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
A dedicated LICENSE file is not needed, files can just contain a SPDX license header at the top, you can find this in all other intree files. Also locally you can use checkpatch to test your changes before submitting.

faxe1008 commented on samples/drivers/rtc/ds1307/src/main.c:
```diff
@@ -0,0 +1,86 @@
+	}
+
+	/* Set the date and time once. */
+	// set_date_time(dev);
```
When submitting check that you did not comment out code. 
Also extracting this into a separate function does not bring any benefit in this case imo.

faxe1008 commented on samples/drivers/rtc/ds1307/src/main.c:
```diff
@@ -0,0 +1,86 @@
+	const struct device *const dev = DEVICE_DT_GET(RTC_NODE);
+
+	/* Check if the RTC is available */
+	if (dev == NULL) {
```
Above macro would error during build time if could not get the pointer, so a check here is not needed.

faxe1008 commented on samples/drivers/rtc/ds1307/src/main.c:
```diff
@@ -0,0 +1,86 @@
+{
+	int32_t ret = 0;
+
+	/* Get the device structure for the RTC */
```
Generally speaking comments are fine for tricky parts but overdoing it produces noise. Getting a device pointer is an operation which is done thousands of times in tree, so comments are not needed.

faxe1008 commented on samples/drivers/rtc/ds1307/README.rst:
```diff
@@ -0,0 +1,33 @@
+DS1307: RTC Module
+#######################################################
```
In RST generally you make the heading specifier just as long as your actual heading.

faxe1008 reviewed (commented):
In general I think having the RTC drivers be build during CI would be a good thing. If this should be a sample or rather a build all test I'd leave to @bjarki-andreasen :^)

Just left some comments. In general you might want to read the contribution guidelines:
https://docs.zephyrproject.org/latest/contribute/guidelines.html

walidbadar committed:
samples: rtc: ds1307: Implementation of RTC API

This sample application periodically reads date and time data from the DS1307 RTC module, and displays on the console.

Signed-off-by: Muhammd Waleed Badar <walid.badar@gmail.com>

walidbadar commented:
@faxe1008  Sorry about the mess, this is my first pull request so bare with me also git does not like windows whitespaces.
Just to clarify it's just implementation of rtc driver not the actual driver.
